applications/StructuralMechanicsApplication/custom_processes/set_moving_load_process.h

```diff
@@ -0,0 +1,216 @@
+
+    typedef std::size_t SizeType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    SetMovingLoadProcess(ModelPart& rModelPart,
+                                  Parameters Parameters);
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * \brief  Initializes the set moving load process. Check if load functions and a velocity function are present in the parameters.
+     * Sort vector of conditions, and find the start position of the moving load, within the conditions vector.
+     */
+    void ExecuteInitialize() override;
+
+    /**
+     * \brief Initialize solution step. Calculate the load based on the load functions if present, else retrieve the load from the input parameters.
+     * Loop over the conditions and find, on which condition the load is located. Then set the load on the condition element, if the load is located
+     * within the element. If the moving load is not located on the condition element, set the load to zero.
+     */
+    void ExecuteInitializeSolutionStep() override;
+
+    /**
+     * \brief Finalizes solution step. Sets load velocity based on load velocity function if present, else load velocity is retrieved from the input values.
+     * Then move the load based on the current position and the load velocity.
+     */
+    void ExecuteFinalizeSolutionStep() override;
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    virtual std::string Info() const override {
+        return "SetMovingLoadProcess";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override {
+        rOStream << "SetMovingLoadProcess";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override {
+    }
+
+    ///@}
+
+private:
+    ///@name Member Variables
+    ///@{
+
+    ModelPart& mrModelPart;
+    Parameters mParameters;
+
+    // vector of sorted conditions
+    std::vector<Condition> mSortedConditions;
+
+    // vector of booleans which indicate if the nodes in the condition are in reversed order with respect to the direction of the moving load
+    std::vector<bool> mIsCondReversedVector;
+
+    // point of origin of the moving load
+    array_1d<double, 3> mOriginPoint;
+
+    // array of directions, of the moving load. This array indicates if the load moves in positive or negative direction with respect to the global
+    // axis
+    array_1d<int,3> mDirection;
+
+    // distance of the load, with respect to the first node of the first condition in the sorted conditions vector
+    double mCurrentDistance;
+
+    // bool which indicates if a load function is used to determine the value of the load
+    bool mUseLoadFunction;
+
+    // bool which indicates of velocity function is used to determine the value of the load velocity
+    bool mUseVelocityFunction;
+
+    // vector of load functions
+    std::vector<BasicGenericFunctionUtility> mLoadFunctions;
```

Comment:
I think you need to implement the `Serializer` in case that you want to use it together with a restart

Reply:
Do you mean that i have to save and load the "mCurrentDistance",(the parameter which changes each time step)  or also the other parameters?

Reply:
Hi @AlejandroCornejo, I've added the serializer. Do you have an idea when you have time to review that part?